include/aikido/planner/vectorfield/VectorFieldPlanner.hpp

```diff
@@ -2,58 +2,88 @@
+      const aikido::constraint::TestablePtr _constraint,
+      double _initialStepSize = 0.1);
 
-/// Callback function of status of planning
-///
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in] _t Planned time of a given duration
-/// \return Status of vectorfield planner
-using VectorFieldStatusCallback = std::function<VectorFieldPlannerStatus(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    double _t)>;
+  /// Vectorfield callback function that returns joint velocities for
+  /// integration.
+  ///
+  /// \param[in] _q Position in configuration space
+  /// \param[out] _qd Joint velocities in configuration space
+  /// \param[in] _t Current time being planned
+  void step(const Eigen::VectorXd& _q, Eigen::VectorXd& _qd, double _t);
 
-/// Plan to a trajectory by a given vector field.
-///
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in] _constraint Trajectory-wide constraint that must be satisfied
-/// \param[in] _timestep How long an evaluation step is
-/// \param[in] _vectorField Callback of vector field calculation
-/// \param[in] _statusCb Callback of planning status
-/// \return Trajectory or \c nullptr if planning failed
-std::unique_ptr<aikido::trajectory::Spline> planPathByVectorField(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    const aikido::constraint::TestablePtr& _constraint,
-    double _timestep,
-    const VectorFieldCallback& _vectorFieldCb,
-    const VectorFieldStatusCallback& _statusCb);
+  /// Check status after every intergration step
+  ///
+  /// \param[in] _q Position in configuration space
+  /// \param[in] _t Current time being planned
+  void check(const Eigen::VectorXd& _q, double _t);
+
+  /// Generate a trajectory following the vector field along given time
+  ///
+  /// \param[in] _integrationTimeInterval Position in configuration space
+  /// \param[in] _timelimit Timelimit for integration calculation
+  /// \param[in] _useCollisionChecking Whether collision checking is
+  /// considered in planning.
+  /// \param[in] _useDofLimitChecking Whether Dof Limits are considered
+  /// in planning
+  /// \return A trajectory following the vector field
+  std::unique_ptr<aikido::trajectory::Spline> followVectorField(
+      double _integrationTimeInterval,
+      double _timelimit,
+      double _useCollisionChecking = true,
+      double _useDofLimitChecking = true);
+
+protected:
+  aikido::planner::vectorfield::ConfigurationSpaceVectorFieldPtr mVectorField;
+  aikido::constraint::TestablePtr mConstraint;
+  aikido::statespace::dart::MetaSkeletonStateSpacePtr mMetaSkeletonStateSpace;
+  dart::dynamics::MetaSkeletonPtr mMetaSkeleton;
+  dart::dynamics::BodyNodePtr mBodyNode;
+  dart::common::Timer mTimer;
+  double mTimelimit;
+
+  std::vector<Knot> mKnots;
+
+  double mInitialStepSize;
+  int mCacheIndex;
+  int mIndex;
+
+  bool mEnableCollisionCheck;
+  bool mEnableDofLimitCheck;
```

Comment:
It seems `mEnableCollisionCheck` and `mEnableDofLimitCheck` are not actually used because they are always overwritten in `followVectorField()`.

I think we should either (1) remove these member variables and always pass them as parameters of `followVectorField()` or (2) add getters and setters for them and remove those parameters from `followVectorField()`. I'm inclined to (2) because those parameters should represent _state_ of the planner but not temporary variables.

If we choose (2) then we might want to set those parameters in the constrcutor.

Reply:
These two member variables are actually used in `check()` and `step()`. These two functions are only internally called inside `followVectorField()`. These two parameters are about how you shall follow a vector field (like integral time), so I put them as input arguments of `followVectorField()`. They are set to member variables so that `check()` and `step()` could access. Any suggestion?